container/container.go

```diff
@@ -1,201 +1,523 @@
-	return NewAttributesFromV2(c.v2.GetAttributes())
+// CreatedAt returns container's creation time set using SetCreationTime.
+//
+// Zero Container has zero timestamp (in seconds).
+func CreatedAt(cnr Container) time.Time {
+	var sec int64
+
+	attr := cnr.Attribute(attributeTimestamp)
+	if attr != "" {
+		var err error
+
+		sec, err = strconv.ParseInt(cnr.Attribute(attributeTimestamp), 10, 64)
+		if err != nil {
+			panic(fmt.Sprintf("parse container timestamp: %v", err))
+		}
+	}
+
+	return time.Unix(sec, 0)
 }
 
-func (c *Container) SetAttributes(v Attributes) {
-	c.v2.SetAttributes(v.ToV2())
+// SetSubnet places the Container on the specified NeoFS subnet. If called,
+// container nodes will only be selected from the given subnet, otherwise from
+// the entire network.
+func SetSubnet(cnr *Container, subNet subnetid.ID) {
+	cnr.SetAttribute(container.SysAttributeSubnet, subNet.EncodeToString())
 }
 
-func (c *Container) PlacementPolicy() *netmap.PlacementPolicy {
-	m := c.v2.GetPlacementPolicy()
-	if m == nil {
-		return nil
+// Subnet return container subnet set using SetSubnet.
+//
+// Zero Container is bound to zero subnet.
+func Subnet(cnr Container) (res subnetid.ID) {
+	val := cnr.Attribute(container.SysAttributeSubnet)
+	if val != "" {
+		err := res.DecodeString(val)
+		if err != nil {
+			panic(fmt.Sprintf("invalid subnet attribute: %s (%v)", val, err))
+		}
 	}
 
-	var p netmap.PlacementPolicy
-	// FIXME(@cthulhu-rider): #225 handle error
-	err := p.ReadFromV2(*m)
-	if err != nil {
-		panic(err)
+	return
+}
+
+const (
+	attributeHomoHashKey     = container.SysAttributePrefix + "DISABLE_HOMOMORPHIC_HASHING"
+	attributeHomoHashEnabled = "true"
+)
+
+// DisableHomomorphicHashing sets flag to disable homomorphic hashing of the
+// Container data.
+//
+// See also IsHomomorphicHashingDisabled.
+func DisableHomomorphicHashing(cnr *Container) {
+	cnr.SetAttribute(attributeHomoHashKey, attributeHomoHashEnabled)
+}
+
+// IsHomomorphicHashingDisabled checks if DisableHomomorphicHashing was called.
+//
+// Zero Container has enabled hashing.
+func IsHomomorphicHashingDisabled(cnr Container) bool {
+	return cnr.Attribute(attributeHomoHashKey) == attributeHomoHashEnabled
+}
+
+// Domain represents information about container domain registered in the NNS
+// contract deployed in the NeoFS network.
+type Domain struct {
+	name, zone string
+}
+
+// SetName sets human-friendly container domain name.
+func (x *Domain) SetName(name string) {
+	x.name = name
+}
+
+// Name returns name set using SetName.
+//
+// Zero Domain has zero name.
+func (x Domain) Name() string {
+	return x.name
+}
+
+// SetZone sets zone which is used as a TLD of a domain name in NNS contract.
+func (x *Domain) SetZone(zone string) {
+	x.zone = zone
+}
+
+// Zone returns domain zone set using SetZone.
+//
+// Zero Domain has "container" zone.
+func (x Domain) Zone() string {
+	if x.zone != "" {
+		return x.zone
 	}
 
-	return &p
+	return "container"
```

Comment:
It is a deploy-time parameter of a contract, may be it is better to return empty here?

Reply:
Oh, right.